conans/client/generators/make.py

```diff
@@ -28,10 +34,46 @@ def content(self):
         for line_as_list in self.create_deps_content():
             content.append("".join(line_as_list))
 
+        additional_content = self.create_additional_content()
+
+        content.append(additional_content)
+        content.append(self.makefile_newline)
         content.append("#-------------------------------------------------------------------#")
         content.append(self.makefile_newline)
         return self.makefile_newline.join(content)
 
+    def create_additional_content(self):
+        additional_content = textwrap.dedent("""
+            CONAN_CPPFLAGS      += $(addprefix -I,$(CONAN_INCLUDE_DIRS))
+            CONAN_CPPFLAGS      += $(addprefix -D,$(CONAN_DEFINES))
+            CONAN_LDFLAGS       += $(addprefix -L,$(CONAN_LIB_DIRS))
+            CONAN_LDFLAGS       += $(CONAN_RPATHFLAGS)
+            CONAN_LDLIBS        += $(addprefix -l,$(CONAN_SYSTEM_LIBS))
+            CONAN_LDLIBS        += $(addprefix -l,$(CONAN_LIBS))
+
+            CONAN_SET_SHARED = {{set_shared}}
+            ifeq ($(CONAN_SET_SHARED),True)
+                CONAN_LDFLAGS += $(CONAN_SHARED_LINKER_FLAGS)
+            else
+                CONAN_LDFLAGS += $(CONAN_EXE_LINKER_FLAGS)
+            endif
+
+            # Call this function in your Makefile to have Conan variables added to standard variables
+            # Example:  $(call CONAN_BASIC_SETUP)
+
+            CONAN_BASIC_SETUP = \\
+                $(eval CFLAGS   += $(CONAN_CFLAGS)) ; \\
+                $(eval CXXFLAGS += $(CONAN_CXXFLAGS)) ; \\
+                $(eval CPPFLAGS += $(CONAN_CPPFLAGS)) ; \\
+                $(eval LDFLAGS  += $(CONAN_LDFLAGS)) ; \\
+                $(eval LDLIBS   += $(CONAN_LDLIBS)) ;
+        """)
+        t = Template(additional_content)
+        context = {
+            "set_shared": True if self.conanfile.options.get_safe("shared") else False,
```

Comment:
This "shared" belongs exclusively to the toolchain, not to the generator. Please remove.

Reply:
I considered that, but this highlights the one major dilemma with separating the TOOLCHAIN from the GENERATOR.  note that this problem does not exist for CMake because CMake distinguishes EXE targets from LIBRARY targets. MSBuild, Make, and potentially others do not. 

The problem is that we need to choose between applying `SHAREDLINKFLAGS` and `EXELINKFLAGS` from dependencies. 

If we don't do this, the next best thing would be to parameterize the `CONAN_BASIC_SETUP` macro instead to choose between `SHARED` and `EXE`.